src/sentry/api/endpoints/organization_events_trace.py

```diff
@@ -1261,150 +1288,81 @@ def serialize_with_spans(
-                if orphan["trace.parent_transaction"] is None
-            ]
-
-            # Build the trees for all the orphan transactions.
-            orphans = self.visit_transactions(
-                orphan_roots,
-                remaining_transactions,
-                errors,
-                visited_transactions,
-                visited_errors,
-            )
+        with sentry_sdk.start_span(op="serialize", description="associate children"):
+            visited_transactions: set[str] = {
+                transaction.event["id"] for transaction in root_traces
+            }
+            for transaction in serialized_transactions:
+                event_id = transaction.event["id"]
+                if event_id in parent_event_map:
+                    children_events = parent_event_map.pop(event_id)
+                    transaction.children = sorted(children_events, key=child_sort_key)
+                if event_id in parent_error_map:
+                    transaction.errors = sorted(
+                        parent_error_map.pop(event_id), key=lambda k: k["timestamp"]
+                    )
 
-        with sentry_sdk.start_span(op="nodestore", description="remaining orphans"):
-            # Remaining are transactions with parent transactions but those
-            # parents don't map to any of the existing transactions.
-            remaining_transactions = self.calculate_remaining_transactions(
-                transactions, visited_transactions
-            )
-            orphans.extend(
-                self.visit_transactions(
-                    remaining_transactions,
-                    remaining_transactions,
-                    errors,
-                    visited_transactions,
-                    visited_errors,
-                )
-            )
+        with sentry_sdk.start_span(op="serialize", description="more orphans"):
+            for transaction in sorted(serialized_transactions, key=child_sort_key):
+                if transaction.event["id"] not in visited_transactions:
+                    if transaction not in orphans:
+                        orphans.append(transaction)
+                    visited_transactions.add(transaction.event["id"])
+                    for child in transaction.children:
+                        visited_transactions.add(child.event["id"])
 
         with sentry_sdk.start_span(op="serialize", description="sort"):
             # Sort the results so they're consistent
-            orphan_errors = sorted(
-                [error for error in errors if error["id"] not in visited_errors],
-                key=lambda k: k["timestamp"],
-            )
+            orphan_errors.sort(key=lambda k: k["timestamp"])
             root_traces.sort(key=child_sort_key)
             orphans.sort(key=child_sort_key)
 
+        visited_transactions = set()
         with sentry_sdk.start_span(op="serialize", description="to dict"):
             return {
```

Comment:
for readability I'd maybe just make a new empty set local to this return instead of reusing the same var as above